### ISSUE TITLE
Synthetic symbol introduced by #12352 is not needed when extending from any is allowed

### ISSUE THREAD

**Issue by sandersn** [Microsoft/TypeScript#15040]
Once #14935 is merged, #12352 is no longer needed. #12352 introduces a new symbol for unknown modules so that `getBaseConstructorTypeOfClass` can give a nice error when trying to extend from an unknown symbol. However, this is now allowed, so there's no reason to have this new symbol.
